tests/adapter/test_json.py

```diff
@@ -1,24 +1,42 @@
 from __future__ import annotations
 
 import json
-from typing import TYPE_CHECKING
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Callable
 
 import pytest
 
 from flow.record import RecordReader, RecordWriter
-from tests._utils import generate_records
+from tests._utils import generate_command_records, generate_records
 
 if TYPE_CHECKING:
+    from collections.abc import Iterator
     from pathlib import Path
 
+    from flow.record import Record
 
-def test_json_adapter(tmp_path: Path) -> None:
+
+class GeneratorType(Enum):
+    RECORDS = auto()
+    COMMAND = auto()
+
+
+FUNCTIONS = {
+    GeneratorType.RECORDS: generate_records,
+    GeneratorType.COMMAND: generate_command_records,
+}
+
+
+@pytest.mark.parametrize("generator_function", FUNCTIONS.keys())
```

Comment:
I can see what you are getting at, but I would prefer a specific test that would then just have fieldtypes parametrized or auto generated.

For now I would just add an extra test in the likes of `test_json_command_fieldtype"